### PR TITLE
Add storybook

### PR DESCRIPTION
**Pull request by RaedsLab**
# TL;DR 

Adds storybook to use as local development environment.
Should close issue https://github.com/ricardo-ch/react-easy-sort/issues/1. 

### Removed existing example 

Removed `examples/src/` and the `webpack` config associated to it. 

### Created one `Example` story

It is the same code from the now deleted `examples` directory. 

### Added husky

Used a hook to make sure we build the storybook automatically.

## Possible improvements

We can load the `README.md`as the default welcome screen for the storybook.

We can automatically generate the latest version of docs with every push. (maybe Github Action?)

### CODE REVIEW

File: .storybook/preview.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+
+export const parameters = {
+  actions: { argTypesRegex: "^on[A-Z].*" },
```
Comment (ValentinH) [inline .storybook/preview.js:3]: what is this for?

File: package.json [hunk 2]
```diff
@@ -31,10 +31,14 @@
...
-    "prepublishOnly": "yarn-or-npm build",
+    "prepublishOnly": "pinst --disable",
     "precommit": "lint-staged",
     "format": "prettier --write src/**/*.ts*",
-    "npm:publish": "np --contents=dist"
+    "npm:publish": "np --contents=dist",
+    "storybook": "start-storybook -p 6006",
+    "build-storybook": "build-storybook -o ./docs",
+    "postinstall": "husky install",
+    "postpublish": "pinst --enable"
```
Comment (ValentinH) [inline package.json:41]: I don't think that we need `pinst` at all because the  `package.json` we publish is a light version of this one without any `scripts`.

File: package.json [hunk 3]
```diff
@@ -31,10 +31,14 @@
     "lint:ts": "tsc --noEmit",
     "prebuild": "rimraf dist",
     "build:copy-files": "babel-node ./scripts/copy-build-files.js",
-    "prepublishOnly": "yarn-or-npm build",
+    "prepublishOnly": "pinst --disable",
```
Comment (ValentinH) [inline package.json:34]: we absolutely need to run `build` before publishing to ensure we publish the latest version.

**Review by ValentinH — CHANGES_REQUESTED** [review]
Thanks for taking care of this!

I don't think the built storybook docs should be committed. As you pointed out in the improvements, it should be done in a GH action. A simpler version could be a manual publishing to Github pages like we do for react-easy-crop [here](https://github.com/ricardo-ch/react-easy-crop/blob/main/docs/package.json#L13).

File: package.json (hunk 3 above)
Comment (RaedsLab) [reply package.json:34]: Oups that was absolutely not intended !

File: .storybook/preview.js (hunk 1 above)
Comment (RaedsLab) [reply .storybook/preview.js:3]: No idea it was generated by storybook's `npx sb init`

Comment (ValentinH) [reply .storybook/preview.js:3]: ok 🙂

File: package.json (hunk 2 above)
Comment (RaedsLab) [reply package.json:41]: Makes sense ! I just followed husky docs blindly, should have looked more.

File: .storybook/preview.js (hunk 1 above)
Comment (RaedsLab) [reply .storybook/preview.js:3]: I'll try to remove it and see if anything breaks 🤷‍♀️

**Comment by ValentinH** [comment]
@RaedsLab GH actions seems quite straighforward actually 🎉 
https://github.com/marketplace/actions/deploy-to-github-pages

File: .storybook/preview.js (hunk 1 above)
Comment (ValentinH) [reply .storybook/preview.js:3]: No this is fine if this is coming from the official installation method 🙂

**Comment by RaedsLab** [comment]
> But I can do it later if you don't have time for it 🙂
@ValentinH That would awesome thanks!

File: .storybook/preview.js (hunk 1 above)
Comment (RaedsLab) [reply .storybook/preview.js:3]: https://storybook.js.org/docs/react/configure/overview#configure-story-rendering

**Review by ValentinH — COMMENTED** [review]
The `docs` folder needs to be added to the `.gitignore` file. Then, it might make the change in `.vscode/settings.json` non-required anymore as VSCode doesn't search in git ignored files by default.

File: package.json [hunk 4]
```diff
@@ -21,9 +21,9 @@
   "license": "MIT",
   "sideEffects": false,
   "scripts": {
-    "start": "webpack-dev-server --mode development",
+    "start": "yarn storybook",
     "version": "yarn && yarn build",
-    "build": "yarn-or-npm bili && yarn-or-npm build:copy-files",
+    "build": "yarn-or-npm bili && yarn-or-npm build:copy-files && yarn build-storybook",
```
[the remark this answers is not in the document]
Comment (RaedsLab) [reply package.json:26]: Makes sense! so maybe I can even remove `/docs`and put it under .gitignore ?

[the remark this answers is not in the document]
Comment (ValentinH) [reply package.json]: Yes, see the main comment or my review 😅

[the remark this answers is not in the document]
Comment (RaedsLab) [reply package.json]: Ah missed that message.. :D

**Review by ValentinH — APPROVED** [review]
